completion/available/cargo.completion.bash

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env bash
+
+# cargo (Rust package manager) completion
+
+source $HOME/.cargo/env
+
+# Required for bash versions < 4.1
+# Default bash version is 3.2 on latest macOS. See #6874
+shopt -s extglob
+
+command -v cargo >/dev/null 2>&1 &&
```

Comment:
If you could please switch this to the `_command_exists` helper function to standardize this test. I suggest something like

```bash
_command_exists cargo || return 0
```

or 

```bash
if ! _command_exists cargo ; then
	return 0
fi
```

This is important because your current approach of using `&&` can return 127 when instead you want to gracefully exit.